modules/EPMC.py

```diff
@@ -0,0 +1,183 @@
+    parser.add_argument('--logFile', help='Destination of the logs generated by this script.', type=str, required=False)
+    parser.add_argument('--local', help='Destination of the logs generated by this script.', action='store_true', required=False, default=False)
+    args = parser.parse_args()
+
+    # extract parameters:  
+    cooccurrenceFile = args.cooccurrenceFile
+
+
+    # Initialize logger based on the provided logfile. 
+    # If no logfile is specified, logs are written to stderr 
+    logging.basicConfig(
+        level=logging.INFO,
+        format='%(asctime)s %(levelname)s %(module)s - %(funcName)s: %(message)s',
+        datefmt='%Y-%m-%d %H:%M:%S',
+    )
+    if args.logFile:
+        logging.config.fileConfig(filename=args.logFile)
+    else:
+        logging.StreamHandler(sys.stderr)
+
+    # Parse output file:
+    out_file = args.outputFile
+
+    ##
+    ## Initialize spark session
+    ##
+    global spark
+    local = True
+
+    if args.local:
+        sparkConf = (SparkConf()
+                 .set("spark.driver.memory", "15g")
+                 .set("spark.executor.memory", "15g")
+                 .set("spark.driver.maxResultSize", "0")
+                 .set("spark.debug.maxToStringFields", "2000")
+                 .set("spark.sql.execution.arrow.maxRecordsPerBatch", "500000")
+                 )
+        spark = (
+            SparkSession.builder
+                .config(conf=sparkConf)
+                .master('local[*]')
+                .getOrCreate()
+        )
+    else:
+        sparkConf = (SparkConf()
+                 .set("spark.driver.maxResultSize", "0")
+                 .set("spark.debug.maxToStringFields", "2000")
+                 .set("spark.sql.execution.arrow.maxRecordsPerBatch", "500000")
+                 )
+        spark = (
+            SparkSession.builder
+                .config(conf=sparkConf)
+                .getOrCreate()
+        )
+    logging.info(f'Spark version: {spark.version}')
+
+    ##
+    ## Log parameters:
+    ##
+    logging.info(f'Cooccurrence file: {cooccurrenceFile}')
+    logging.info(f'Output file: {out_file}')
+    logging.info(f'Generating evidence:')
+
+    filtered_target_terms = ["TEC", "TECS", "Tec", "tec", "'", "(", ")", "-", "-S", "S", "S-", "SS", "SSS",
+         "Ss", "Ss-", "s", "s-", "ss", "U3", "U6", "u6", "SNORA70", "U2", "U8"]
+
+    ##
+    ## Load/filter datasets
+    ##
+    # partitionKeys = ['pmid', 'targetFromSourceId', 'diseaseFromSourceMappedId']
+    # w = Window.partitionBy(*partitionKeys)
+    filtered_cooccurrence_df = (
+        # Reading file:
+        spark.read.parquet(cooccurrenceFile)
+
+        # Filtering for diases/target cooccurrences:
+        .filter(
+            (col('type') == "GP-DS") & 
+            (col('isMapped') == True) & 
+            (col('pmid').isNotNull()) & 
+            (length(col('text')) < 600) &
```

Comment:
The rest of the filters kind of make sense intuitively but not this one. What does it do? Why don't we want records with text longer than 600? And why exactly 600? :)

Reply:
The threshold is quite arbitrary. The distribution of the length of the sentences is extremely skewed to the left: the median length of the 61M sentences is 230, while the mean length is 900. This extreme skewness is mostly due to problems in splitting the text into sentences. 

I picked 600 as a kind of good-enough threshold where (based on a random sample) 15% of the sentences were genuine sentences while the rest were improperly split sentences or concatenated lists from the paper. By excluding sentences above 600 we are losing ~8000 sentences representing 0.01% of all sentences.

Reply:
Thank you for the explanation! Do you think we could add it to the script, perhaps as a comment in this particular place where `600` first appears?